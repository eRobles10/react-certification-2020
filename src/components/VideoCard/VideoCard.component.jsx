import React from 'react';
import styled from 'styled-components';

const CardContent = styled.div`
    overflow: hidden;
    border-radius: 2px;
    border: 1px solid rgb(236, 236, 236 ) !important;
    display:block;
    width:350px;
    height: 350px;
    margin:15px;

    &:hover{
        background-color: rgb(227 227 227);
    }

`;

const Preview = styled.div`
    display: block;
    height: 180px;
`;

const Details = styled.div`
    display: block;
    padding 15px;
`;

const Thumbnail = styled.img`
    width: 350px;
    height: 180px;

    @media only screen and (max-width: 600px) {
        width:30%;
    }
`;

const VideoButton = styled.button`
    background: none;
    width: 100%;
    display: block;
    padding:0;
    margin:0;
    align-items:center;
    border:0;
    cursor:pointer;
    user-select:none;
    outline:none;
    text-align:left;

`



const VideoCard =(props) =>{

    const showDetail=(e)=>{
        e.preventDefault();
        props.setShowDetail({id:props.id.videoId,
            title:props.snippet.title,
            description:props.snippet.description, setShowDetail:props.setShowDetail})
    }

    return (
        <CardContent className="Video-Card" >
            <VideoButton type="button"  onClick={showDetail} >
                <Preview >
                    <Thumbnail alt="default" src={props.snippet.thumbnails.high.url} />
                </Preview>
                <Details>
                    <h3>{props.snippet.title}</h3>
                    <p>{props.snippet.description}</p>
                </Details>
            </VideoButton>
        </CardContent>

    );
}

export default VideoCard;
