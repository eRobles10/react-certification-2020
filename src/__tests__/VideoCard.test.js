import React from 'react';
import renderer from 'react-test-renderer';
import VideoCard from '../components/VideoCard';


describe('VideoCard Test', () => {
    const id = {videId:1};
    const thumbnail= {high:"prueba.jpg"};
    const snippet={title:"Test", description:"Test Description", thumbnails:thumbnail};
    const  setShowDetail = ()=> {

    }

    test("VideoCard is getting rendered", () => {
        const videoCard = renderer
                .create(<VideoCard id={id} snippet={snippet} setShowDetail={setShowDetail} />)
                .toJSON();

        expect(videoCard).toMatchSnapshot();

    });


    test("Ensure that Video Card is rendered", () => {


    });

    test("Ensure that Thumbnail in Video Card match with the one in the mock item", () => {


    });

    test("Ensure that Title in Video Card match with the mock item selected", () => {


    });

    test("Ensure that Title in Video Card match with the mock item selected", () => {


    });

});