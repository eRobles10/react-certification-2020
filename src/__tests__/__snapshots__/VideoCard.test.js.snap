// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard Test VideoCard is getting rendered 1`] = `
<div
  className="sc-bdfBwQ gTyMyW Video-Card"
>
  <button
    className="sc-eCssSg foQmNZ"
    onClick={[Function]}
    type="button"
  >
    <div
      className="sc-gsTCUz KZrsn"
    >
      <img
        alt="default"
        className="sc-hKgILt cbCdcf"
      />
    </div>
    <div
      className="sc-dlfnbm liHOfl"
    >
      <h3>
        Test
      </h3>
      <p>
        Test Description
      </p>
    </div>
  </button>
</div>
`;
